src/lib.rs

```diff
@@ -345,12 +346,27 @@ pub mod pallet {
 
 			let current_initialized_rewards = InitializedRewardAmount::<T>::get();
 
-			// This ensures that the rewards match the pot
+			let reward_difference = Self::pot().saturating_sub(current_initialized_rewards);
+
+			// Ensure the difference is not bigger than the total number of contributors
 			ensure!(
-				current_initialized_rewards == Self::pot(),
+				reward_difference < TotalContributors::<T>::get().into(),
 				Error::<T>::RewardsDoNotMatchFund
 			);
 
+			if reward_difference > 0u32.into() {
+				let to_burn = T::RewardCurrency::burn(reward_difference);
+
+				// Shouldnt fail, as the fund should be enough to burn and nothing is locked
+				T::RewardCurrency::settle(
+					&PALLET_ID.into_account(),
+					to_burn,
+					WithdrawReasons::TRANSFER,
+					AllowDeath,
+				)
+				.map_err(|_| Error::<T>::ErrorBurningImbalance)?;
```

Comment:
I've never seen this method of burning before. But looking at the docs, I think it is correct. I've previously seen the technique of calling [withdraw](https://crates.parity.io/frame_support/traits/tokens/currency/trait.Currency.html#tymethod.withdraw) and then dropping the imbalance which will automatically adjust the total issuance. But honestly I don't see either one as better than the other, so I'm fine leaving it.

Reply:
I ended up using withdraw, as with the expect was giving less problems (settle returns an imbalance that does not have fmt derived)